### PR TITLE
fine tuning job listing

### PR DESCRIPTION
**Pull request by BrendanMartin**
**Describe the change**
Added support for the listing of fine-tuning jobs.

**Describe your solution**
Essentially copied and modified the fine-tuning events listing code that was already present.

**Tests**
Added unit test.

**Additional context**
Add any other context or screenshots or logs about your pull request here. If the pull request relates to an open issue, please link to it.

Issue: #488

### CODE REVIEW

**Comment by BrendanMartin** [comment]
I'm not sure why coverage is lower. I used the same code as FineTuningJobEvent.

**Comment by sashabaranov** [comment]
@BrendanMartin Hey, thank you for the PR! Please throw in a test case here, it should fix the coverage issue: https://github.com/sashabaranov/go-openai/blob/master/client_test.go#L195

File: fine_tuning_job.go [hunk 1]
```diff
@@ -151,3 +157,61 @@ func (c *Client) ListFineTuningJobEvents(
 	err = c.sendRequest(req, &response)
 	return
 }
+
+type listFineTuningJobsParameters struct {
+	after *string
+	limit *int
+}
+
+type ListFineTuningJobsParameter func(*listFineTuningJobsParameters)
```
Comment (sashabaranov) [inline fine_tuning_job.go:166]: Should we unify all our listing setters? (Maybe even with a small-ish API break?) 

@BrendanMartin Would you have the bandwidth to do this unification? (No worries if not, we can merge and update this afterward!)

@vvatanabe would love to hear your thoughts here.

The other listing setters: https://github.com/sashabaranov/go-openai/blob/master/fine_tuning_job.go#L100

Comment (BrendanMartin) [reply fine_tuning_job.go:166]: Unfortunately, I can't tackle this at the moment. To be honest, I don't even know what unification would look like.

Comment (sashabaranov) [reply fine_tuning_job.go:166]: @BrendanMartin got it, thanks you!

Comment (vvatanabe) [reply fine_tuning_job.go:166]: @sashabaranov Sorry for the late reply! I'd like to consolidate using a concise and generic structure like the one below. It's clear at a glance that it's for pagination purposes.

```
type PaginationQuery struct {
    After *string
    Limit *int
}
```

If we're going with this design, I can create a new PR after we merge this one.

Comment (sashabaranov) [reply fine_tuning_job.go:166]: Merged into a non-master branch, let's take our time and unify the pagination approach

**Pull request merged**